src/pymatgen/io/lobster/outputs.py

```diff
@@ -1710,32 +1710,33 @@ def has_good_quality_check_occupied_bands(
         Returns:
             bool: True if the quality of the projection is good.
         """
-        for matrix in self.band_overlaps_dict[Spin.up]["matrices"]:
-            for iband1, band1 in enumerate(matrix):
-                for iband2, band2 in enumerate(band1):
-                    if iband1 < number_occ_bands_spin_up and iband2 < number_occ_bands_spin_up:
-                        if iband1 == iband2:
-                            if abs(band2 - 1.0).all() > limit_deviation:
-                                return False
-                        elif band2.all() > limit_deviation:
-                            return False
-
-        if spin_polarized:
-            for matrix in self.band_overlaps_dict[Spin.down]["matrices"]:
-                for iband1, band1 in enumerate(matrix):
-                    for iband2, band2 in enumerate(band1):
-                        if number_occ_bands_spin_down is None:
-                            raise ValueError("number_occ_bands_spin_down has to be specified")
-
-                        if iband1 < number_occ_bands_spin_down and iband2 < number_occ_bands_spin_down:
-                            if iband1 == iband2:
-                                if abs(band2 - 1.0).all() > limit_deviation:
-                                    return False
-                            elif band2.all() > limit_deviation:
-                                return False
+        if spin_polarized and number_occ_bands_spin_down is None:
+            raise ValueError("number_occ_bands_spin_down has to be specified")
+
+        for spin in (Spin.up, Spin.down) if spin_polarized else (Spin.up,):
+            num_occ_bands = number_occ_bands_spin_up if spin is Spin.up else number_occ_bands_spin_down
+
+            for overlap_matrix in self.band_overlaps_dict[spin]["matrices"]:
+                sub_array = self.get_sub_array(num_occ_bands=num_occ_bands, overlap_matrix=overlap_matrix)  # type: ignore[arg-type]
+
+                if not np.allclose(sub_array, np.identity(num_occ_bands), atol=limit_deviation, rtol=0):
+                    return False
 
         return True
 
+    @staticmethod
+    def get_sub_array(num_occ_bands: int, overlap_matrix: np.ndarray) -> np.ndarray:
```

Comment:
Would it be overkill to make this a separate method?

Reply:
It's purely personal taste, but I'm feeling having a method just for `return np.asarray(overlap_matrix)[:num_occ_bands, :num_occ_bands]` might be too overkill? I.e. we could just do this inline?

Reply:
oki, I will get rid of this method and the associated test then

Reply:
Thanks for the change, basically adding a test here would be testing whether slicing a `np.array` works correctly or not, I guess it's not really necessary